docs/language_specification/statements/reset_instruction.md

```diff
@@ -0,0 +1,67 @@
+A **`reset`** instruction performs a measurement plus a Pauli X on the qubit argument.
+The general form of a reset instruction is as follows:
+
+!!! info ""
+
+    &emsp;**`reset`** _qubit-argument_
+
+??? info "Grammar for reset instruction"
+    
+    _reset_:  
+    &emsp; __`reset`__  
+    &emsp; __`reset`__ _qubit-argument_
+
+    _qubit-argument_:  
+    &emsp; _qubit-variable_  
+    &emsp; _qubit-index_
+
+    _qubit-variable_:  
+    &emsp; _identifier_
+
+    _qubit-index_:  
+    &emsp; _index_
+
+!!! example
+    
+    === "Reset of all the qubits"
+    
+        ```linenums="1", hl_lines="3"
+        qubit q
+        qubit[5] qq
+        reset
+        ```
+    
+    === "Reset of a single qubit"
+    
+        ```linenums="1", hl_lines="2"
+        qubit q
+        reset q
+        ```
+    
+    === "Reset of multiple qubits through their register index"
+    
+        ```linenums="1", hl_lines="2"
+        qubit[5] q
+        reset q[2, 4]
+        ```
+
+!!! note
+
+    The reset instruction accepts [SGMQ notation](gates.md#single-gate-multi-qubit-sgmq-notation), similar to gates.
+
+The following code snippet shows how the reset instruction might be used in context.
+
+```linenums="1" hl_lines="8"
+version 3.0
+
+qubit[2] q
+
+H q[0]
+CNOT q[0], q[1]
+
+reset q  // Measurement in the standard basis, plus a Pauli X on q
+```
+
+On the last line of this simple cQASM program,
+the respective states of both qubits in the qubit register are measured along the standard/computational basis.
+Then a Pauli X is performed on both qubits.
```

Comment:
OK, but I'm then adding as well a measure instruction at the end of the program. Because I see that, what you are trying to explain here, is how `reset` interacts with `measure`.

Also, I wouldn't use terms such as _big-endian_ in this documentation. I would just say, for example for a string such as 01101100, if the lowest qubit indices go at the rightest or leftest positions.

Reply:
Also, I wouldn't use terms such as _big-endian_ in this documentation. I would just say, for example for a string such as 01101100, that the lower qubit indices go at the right and the higher at the left.

This note could also be added on the General Overview page, in a section similar to _About the grammar sections_ (something like _About the state strings_). Maybe renaming them to: _How to read the grammar sections_, and _How to read the state strings_.

Reply:
Done.